lib/collection.js

```diff
@@ -2087,17 +2110,36 @@ Collection.prototype.findOneAndUpdate = function(filter, update, options, callba
  * @return {Promise} returns Promise if no callback passed
  * @deprecated use findOneAndUpdate, findOneAndReplace or findOneAndDelete instead
  */
+const findAndModifySchema = {
+  w: { type: ['number', 'string'] },
+  wtimeout: { type: 'number' },
+  j: { type: 'boolean' },
+  remove: { type: 'boolean', default: false },
+  upsert: { type: 'boolean', default: false },
+  new: { type: 'boolean', default: false },
+  projection: { type: 'object' },
+  fields: { type: 'object' },
+  session: { type: ClientSession },
+  arrayFilters: { type: 'array' },
+  readPreference: { overrideOnly: true },
+  serializeFunctions: { type: 'function' },
+  checkKeys: { overrideOnly: true }
+};
 Collection.prototype.findAndModify = deprecate(function(query, sort, doc, options, callback) {
   const args = Array.prototype.slice.call(arguments, 1);
   callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
   sort = args.length ? args.shift() || [] : [];
   doc = args.length ? args.shift() : null;
   options = args.length ? args.shift() || {} : {};
 
-  // Clone options
-  options = Object.assign({}, options);
-  // Force read preference primary
-  options.readPreference = ReadPreference.primary;
+  // Validate options
+  options = validate(
+    findAndModifySchema,
+    options,
+    { serializeFunctions: this.s.serializeFunctions },
```

Comment:
in the operation this was `options.serializeFunctions || coll.s.serializeFunctions`, sounds like another "inheritable property" to me

Reply:
Actually, we already have NODE-1661 for this so I just added this case as a comment to that ticket.